Lib/test/test_fstring.py

```diff
@@ -931,7 +925,8 @@ def test_lambda(self):
 
         # lambda doesn't work without parens, because the colon
         #  makes the parser think it's a format_spec
-        self.assertAllRaise(SyntaxError, 'invalid syntax',
+        self.assertAllRaise(SyntaxError, 
+                            "f-string: expecting a valid expression after '{'",
```

Comment:
Let's special-case lambda and display a message like `f-string: lambda expression are not allowed without parentheses`, since lambda theoretically is a valid expression and the PEP explicitly mentions it.